### ISSUE TITLE
Point Mutation doesn't mutate anything and instead Reproduces

### ISSUE THREAD

**Issue by wulfihm** [trevorstephens/gplearn#124]
When the individual that is being mutated is small enough (nodes) and the point mutation probability (p_point_replace) is set low enough the operator has a high chance of not mutating anything and instead simply reproduces the individual.
This is counterproductive resulting in many copies of the same individual in the population in some cases. If someone wants to have a high reproducing chance one should set the sum of p_crossover, p_subtree_mutation, p_hoist_mutation and p_point_mutation lower than 1.0.
Instead the results are kinda skewed because point mutation does reproduce.

One simple solution would be to always have at least one node that is being mutated (for example if the sum of picked nodes is zero pick one at random).
See def point_mutation(self, random_state) in _program.py

However if we did that the probability of p_point_replace wouldn't be exactly of that what we defined it as i.e. it would be a little higher than p_point_replace for every individual that is smaller than n nodes. But I think this would be preferred over a method that simply reproduces individuals.

**Comment by wulfihm** [comment]
I created a pull request but noticed that in my experiments that the results are worse than before and creates unpredictable behaviour.
My assumption is that with my pullrequest (which always at least mutated one node when point mutation was selected) that small programs have changes added to them which are too significant in the results they produce. Therefore it seems better to not changed small programs and just let them reproduce.

Therefore I closed my pullrequest and this issue is closed as well.
